### DISCUSSION TITLE
ripgrep in docker

### DISCUSSION THREAD

**Discussion by lpcy** [BurntSushi/ripgrep#2082] [Q&A] [answered]
I wanna use ripgrep in docker container, but the memory usage confuse me. I observed that the tool occupy 100m~1G memory when use default -j parameter or -j 2 at least, and occupy about 4m memory when use -j 1 in same file directory. The huge memory occupy may cause the application of container OOM-killed due to usual memory-limit. Now I think it's useful to force use -j 1, but what else way can I solve the problem properly?

**Comment by BurntSushi** [comment, marked as answer]
I think using `-j1` if you want lower memory usage is entirely appropriate. Basically, it spins up no additional threads. Adding threads adds memory, and not just from the thread itself. But a whole bunch of stuff is copied to each thread. And since multiple threads means parallelism and ripgrep doesn't want to interleave output, it has to buffer the entire contents of a search in memory before printing it. But that should only impact you if your search has a lot of matches.

**Reply by lpcy** [reply]
OK，thanks. I also want to know what way can I enforce use -j 1 in docker without recompile the whole tool source? Have any environment variable to change the default value of -j?

**Reply by BurntSushi** [reply]
You can add a config file with `-j1` in it and set `RIPGREP_CONFIG_PATH` to that file. But you can't _force_ it. Someone can always add `-j8` to increase it. There is no way to force single threaded without making a patch to the code.

**Reply by lpcy** [reply]
Thanks, this is what I want.
